pkg/daemon/cleanup/cleanup.go

```diff
@@ -131,18 +133,18 @@ func getMergedCMData(ctx context.Context, cli client.Client,
 	if err != nil {
 		klog.Infof("error getting services ConfigMap associated with version annotation, "+
 			"falling back to use latest services ConfigMap: %s", err)
-		return latestCMData, nil
+		return latestCMData, true, nil
 	}
 	// If the instance was configured using latestCM, return that
 	if versionCM.GetName() == latestCM.GetName() {
-		return latestCMData, nil
+		return latestCMData, false, nil
 	}
 	mergedServices := mergeServices(latestCMData.Services, versionCMData.Services)
 	mergedEnvVars := merge(latestCMData.WatchedEnvironmentVars, versionCMData.WatchedEnvironmentVars)
 	return &servicescm.Data{
 		Services:               mergedServices,
 		WatchedEnvironmentVars: mergedEnvVars,
-	}, nil
+	}, true, nil
```

Comment:
Why is a best effort cleanup needed here?

Reply:
addressed with a clearer function doc

Reply:
Going off the doc the return value should be flipped.
True should be returned only if the instance was configured using latestCM